code/Terrain/FireHelper.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace Grubs;
+
+public static class FireHelper
+{
+	public static void StartFiresAt( Vector3 position, Vector3 moveDirection, int fireQuantity )
+	{
+		Game.AssertServer();
+
+		for ( var i = 0; i < fireQuantity; i++ )
+		{
+			var baseDirection = Vector3.Random.WithY( 0f ) * 45f;
+			_ = new FireEntity(
+				position + Vector3.Random.WithY( 0f ) * 45f,
+				baseDirection * moveDirection );
+		}
+	}
+}
+
+[Category( "Weapons" )]
+public class FireEntity : ModelEntity, IResolvable
+{
+	public bool Resolved => Time.Now > _expiryTime;
+	private Vector3 _moveDirection { get; set; }
+
+	private readonly float _expiryTime;
+	private const float fireSize = 10f;
+
+	private Particles FireParticle { get; set; }
+
+	public FireEntity()
+	{
+
+	}
+
+	public FireEntity( Vector3 startPosition, Vector3 moveDirection )
+	{
+		Position = startPosition;
+		_moveDirection = moveDirection;
+
+		_expiryTime = Time.Now + Game.Random.Float( 0.5f, 2.5f );
+	}
+
+	public override void Spawn()
+	{
+		FireParticle = Particles.Create( "particles/fire/fire_base.vpcf", this, true );
+		Tags.Add( "fire" );
+		Name = "fire";
+	}
+
+	[GameEvent.Tick.Server]
```

Comment:
Will we have any problems moving the fire only on the server? Would it make sense to throw it into simulate?

Reply:
Works totally fine on the server - the movement is pretty incremental anyway for whatever reason and the move controller does not get affected by fire so it shouldn't affect prediction.